include/dol/s/s_StateInterfaces.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+/// @brief The interface for state IDs.
+/// @details A state ID is made up of a name string and a unique number, where 0 denotes a null state.
+class sStateIDIf_c {
+public:
+    virtual ~sStateIDIf_c() {}
+    virtual bool isNull() const = 0; ///< Returns if this is a null state.
+    virtual bool isEqual(const sStateIDIf_c &other) const = 0; ///< Returns true if both states have the same number.
+
+    virtual bool operator==(const sStateIDIf_c &other) const = 0; ///< Overloaded equality operator, uses ::isEqual.
+    virtual bool operator!=(const sStateIDIf_c &other) const = 0; ///< Overloaded inequality operator, uses ::isEqual.
+
+    virtual bool isSameName(const char *name) const = 0; ///< Returns true if this state ID's is called @p name.
+    virtual const char *name() const = 0; ///< Returns the name of this state ID.
+    virtual unsigned int number() const = 0; ///< Returns the number of this state ID.
+};
+
+/// @brief The interface for state ID checkers.
+/// @details [This class is not really used, but it seems to be intended for testing if a state is "normal", but not sure how this would be useful.]
```

Comment:
```suggestion
/// @brief An interface for state ID checkers.
/// @details [This class is not really used, but it seems to be intended for testing if a state is "normal", but not sure how this would be useful].
```